src/helpers.js

```diff
@@ -0,0 +1,152 @@
+var childProcess = require('child_process');
+var os = require('os');
+var osName = require('os-name');
+var which = require('which');
+
+function run(cmd) {
+  return (childProcess
+    .execSync(cmd, {
+      stdio: [0, 'pipe', 'ignore'],
+    })
+    .toString() || ''
+  ).trim();
+}
+
+function getAndroidStudioVersion() {
+  var androidStudioVersion = 'Not Found';
+  if (process.platform === 'darwin') {
+    try {
+      androidStudioVersion = run(
+        [
+          '/usr/libexec/PlistBuddy',
+          '-c',
+          'Print:CFBundleShortVersionString',
+          '-c',
+          'Print:CFBundleVersion',
+          '/Applications/Android\\ Studio.app/Contents/Info.plist',
+        ].join(' ')
+      )
+        .split('\n')
+        .join(' ');
+    } catch (err) {
+      androidStudioVersion = 'Not Found';
+    }
+  } else if (process.platform === 'linux') {
+    try {
+      var linuxBuildNumber = run('cat /opt/android-studio/build.txt');
+      var linuxVersion = run(
+        'cat /opt/android-studio/bin/studio.sh | grep "$Home/.AndroidStudio" | head -1'
+      ).match(/\d\.\d/)[0];
+      androidStudioVersion = `${linuxVersion} ${linuxBuildNumber}`;
+    } catch (err) {
+      androidStudioVersion = 'Not Found';
+    }
+  } else if (process.platform.startsWith('win')) {
+    try {
+      var windowsVersion = run(
+        'wmic datafile where name="C:\\\\Program Files\\\\Android\\\\Android Studio\\\\bin\\\\studio.exe" get Version'
+      ).replace(/(\r\n|\n|\r)/gm, '');
+      var windowsBuildNumber = run(
+        'type "C:\\\\Program Files\\\\Android\\\\Android Studio\\\\build.txt"'
+      ).replace(/(\r\n|\n|\r)/gm, '');
+      androidStudioVersion = `${windowsVersion} ${windowsBuildNumber}`;
+    } catch (err) {
+      androidStudioVersion = 'Not Found';
+    }
+  }
+  return androidStudioVersion;
```

Comment:
^ WOAH!  THIS IS GOLD!

Reply:
Actually I thought it was garbage, but damn, it works.